### PR TITLE
add util function to update named-ports for an instance group

### PR DESCRIPTION
**Pull request by emicklei**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -173,6 +174,18 @@ Use this command with care!.
...
+## GCP utilities
+
+### util create-named-port [instance-group] [name:[port]
+
+The Cloud SDK has a command to [set-named-ports](https://cloud.google.com/sdk/gcloud/reference/compute/instance-groups/set-named-ports) but not command to add or delete a single name:port mapping. To simplify the migration command for creating a name:port mapping, this gmig util command is added.
+First it calls `get-named-ports` to retrieve all existing mappings. Then it will call `set-named-ports` with the new mapping unless it already exists.
+
+### util delete-named-port [instance-group] [name:[port]
+
+The Cloud SDK has a command to [set-named-ports](https://cloud.google.com/sdk/gcloud/reference/compute/instance-groups/set-named-ports) but not command to add or delete a single name:port mapping. To simplify the migration command for deleting a name:port mapping, this gmig util command is added.
```
Comment (joeyroosing) [inline README.md]: "but not command" should be either "but not a command" or "but no command"

File: command_namedport.go [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+	if len(instanceGroup) == 0 {
+		return fmt.Errorf("missing Compute Instance Group command argument")
+	}
+	parts := strings.Split(c.Args()[1], ":")
+	if len(parts) == 0 {
+		return fmt.Errorf("invalid NAME:PORT command argument")
+	}
+	port, err := strconv.Atoi(parts[1])
+	if err != nil {
+		return fmt.Errorf("PORT must be a positive integer:%v", err)
```
Comment (joeyroosing) [inline command_namedport.go]: Afaik, this is incorrect, Atoi supports negative integers (https://play.golang.org/p/3m-NlgsIeS3)

File: command_namedport.go [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+	if verbose {
+		log.Println(strings.Join(append([]string{"gcloud"}, args...), " "))
+	}
+	data, err := runCommand(cmd)
+	if err != nil {
+		log.Println(string(data)) // stderr
+		return fmt.Errorf("failed to run gcloud get-named-ports: %v", err)
+	}
+	// parse named ports
+	var list []namedPort
```
Comment (joeyroosing) [inline command_namedport.go]: Can this whole part be extracted to another method? Or create a switch and call different methods based on the action which returns the list, would make the code more readable (personal opinion)

File: command_namedport.go (hunk 2 above)
Comment (emicklei) [reply command_namedport.go]: yes but ports not

**Pull request merged**